### PR TITLE
Dynamic Unet future work

### PR DESCRIPTION
**Pull request by KeremTurgutlu**
I've added comments about only supporting size change of H, W -> H/2, W/2 in encoder part and a TODO in order to make this adaptive for custom encoders that might have a different size decay.

### CODE REVIEW

File: fastai/models/unet.py [hunk 1]
```diff
@@ -27,9 +27,10 @@ def conv_bn_relu(in_c, out_c, kernel_size, stride, padding):
 
 
 class UnetBlock(nn.Module):
+    #TODO: ADAPT KERNEL SIZE, STRIDE AND PADDING SO THAT ANY SIZE DECAY WILL BE SUPPORTED
     def __init__(self, up_in_c, x_in_c):
         super().__init__()
-        self.upconv = nn.ConvTranspose2d(up_in_c, up_in_c // 2, 2, 2)
+        self.upconv = nn.ConvTranspose2d(up_in_c, up_in_c // 2, 2, 2) # H, W -> 2H, 2W
```
Comment (apiltamang) [inline fastai/models/unet.py]: Only if you'd be interested ...
I briefly brought this issue up in the forums as well, and I think @jph00 is likely warm towards this: would you mind sharing more info on what _**up_in_c, x_in_c**_ are? Are they tensors, arrays, floats, ints ? And if applicable, what would the generalized shape of these vars  be?

I look at them and I know nothing, nada ... and I assume that being in the thick of things, you're pretty fresh on the material.

Much thanks!

Comment (KeremTurgutlu) [reply fastai/models/unet.py]: up_in_c: channel size of the tensor that is been upsampled.
x_in_c: channel size of the tensor that is been concatenated to the upsampled tensor.

You may go through CaravanaUnet notebook in lectures and get a better intuition about the overall process as well as reading Unet paper might help for the decoding part.

Thanks!

File: fastai/models/unet.py [hunk 2]
```diff
@@ -27,9 +27,10 @@ def conv_bn_relu(in_c, out_c, kernel_size, stride, padding):
 
 
 class UnetBlock(nn.Module):
+    #TODO: ADAPT KERNEL SIZE, STRIDE AND PADDING SO THAT ANY SIZE DECAY WILL BE SUPPORTED
```
Comment (jph00) [inline fastai/models/unet.py]: What's with THE YELLING? ;)

Comment (KeremTurgutlu) [reply fastai/models/unet.py]: Sorry :)

File: fastai/models/unet.py (hunk 1 above)
Comment (apiltamang) [reply fastai/models/unet.py]: True ...
I figured if I said it enough, the word would spread around.
But then, I think it's also by habit. I deal with too many PRs at work where these sorta things keep getting repeated over and over ...

I apologize I haven't been leading by example, though. I really would like to. For the most part, I'm just trying to catch up with the lessons, transitioning to a new job, and just those sorta things. I feel like the PR comments are the least  I can still do, at least for the moment.

Cheers .... and good night!

**Pull request merged**